RileyLinkKitTests/PumpOpsSynchronousTests.swift

```diff
@@ -73,202 +82,433 @@ class PumpOpsSynchronousTests: XCTestCase {
         
         let afterBatteryEventDate = batteryEvent.timestamp.date!.addingTimeInterval(TimeInterval(hours: 10))
         
-        let (_, hasMoreEvents) = sut.convertPumpEventToTimestampedEvents(pumpEvents: pumpEvents, startDate: afterBatteryEventDate, pumpModel: pumpModel)
+        let (_, hasMoreEvents, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: pumpEvents, startDate: afterBatteryEventDate, pumpModel: pumpModel)
         
         XCTAssertFalse(hasMoreEvents)
     }
     
     func testBatteryEventIsCreated() {
         let pumpEvents: [PumpEvent] = [createBatteryEvent()]
         
-        let (events, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: pumpEvents, startDate: Date.distantPast, pumpModel: pumpModel)
+        let (events, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: pumpEvents, startDate: Date.distantPast, pumpModel: pumpModel)
         
         XCTAssertEqual(events.count, 1)
     }
     
     func testMultipleBatteryEvent() {
+        let batteryEvent2007 = createBatteryEvent(withDateComponent: dateComponents2007)
```

Comment:
Testing multiple events tests the logic around the for loop (or any construct used in the future) to make sure multiple events are included and not skipped somehow. It would be an obvious next test to write in a TDD style workflow. 

You could argue the previous test (testBatteryEventIsCreated) that checks a single value is now redundant and could be deleted. I would keep both tests, but would suggest deleting the first one if you think these are too many.